src/server/game/Entities/Object/Object.cpp

```diff
@@ -2539,6 +2539,9 @@ void WorldObject::GetNearPoint(WorldObject const* searcher, float &x, float &y,
         GetNearPoint2D(x, y, distance2d + searcher_size, absAngle + angle);
         z = GetPositionZ();
         UpdateAllowedPositionZ(x, y, z);
+        if (controlZ && fabsf(GetPositionZ() - z) > controlZ)
```

Comment:
did you mean `if (controlZ > 0 && fabsf(GetPositionZ() - z) > controlZ)` ? because `controlZ` is a float

Reply:
For C++ standart it's normal boolean. 
For this condition important only controlZ != 0 (not zero, not 0.0f). 
if rly, idk, it's possible to found place in WoW with Z < 0 - if yes - it's normal for this condition.